### ISSUE TITLE
Bug: whatsapp ignores Theme settings at launch

### ISSUE THREAD

**Issue by fgurdradee** [ferdium/ferdium-app#490]
### Avoid duplicates

- [X] I have searched the [issues tracker](https://github.com/ferdium/ferdium-app/issues) for a bug report similar to mine, in vain

### Ferdium Version

6.0.0-nightly.101

### What Operating System are you using?

Other Linux

### Operating System Version

Linux Mint 20.3 kernel: 5.15.0-41-generic #44~20.04.1-Ubuntu SMP Fri Jun 24 13:27:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### What arch are you using?

x64

### Expected Behavior

Whatsapp should launch in dark mode when Whatsapp - Settings - Theme - Dark has been previously selected.

### Actual Behavior

when launching ferdi - whatsapp loads in light mode. 

### Steps to reproduce

1. launch Ferdi
2. whatsapp loads in light mode
3. whatsapp - settings - Theme - dark is preselected from the previous time i launched the app
4.  whatsapp - settings - Theme - select Light or System default. then OK
5.  go back to main whatsapp screen then select whatsapp - settings - Theme - dark and OK ... to turn back dark mode

### Debug link

{"host":{"platform":"linux","release":"5.15.0-41-generic","screens":[{"id":21442530474005588,"bounds":{"x":1920,"y":6,"width":1920,"height":1080},"workArea":{"x":1920,"y":6,"width":1920,"height":1040},"accelerometerSupport":"unknown","monochrome":false,"colorDepth":24,"colorSpace":"{primaries_d50_referred: [[0.6485, 0.3383],  [0.3351, 0.6182],  [0.1684, 0.0658]], transfer:SRGB, matrix:RGB, range:FULL}","depthPerComponent":8,"size":{"width":1920,"height":1080},"displayFrequency":60,"workAreaSize":{"width":1920,"height":1040},"scaleFactor":1,"rotation":0,"internal":false,"touchSupport":"unknown"},{"id":1657942003464788,"bounds":{"x":3840,"y":0,"width":1920,"height":1080},"workArea":{"x":3840,"y":0,"width":1920,"height":1080},"accelerometerSupport":"unknown","monochrome":false,"colorDepth":24,"colorSpace":"{primaries:BT709, transfer:SRGB, matrix:RGB, range:FULL}","depthPerComponent":8,"size":{"width":1920,"height":1080},"displayFrequency":60,"workAreaSize":{"width":1920,"height":1080},"scaleFactor":1,"rotation":0,"internal":false,"touchSupport":"unknown"},{"id":1251633106950741,"bounds":{"x":0,"y":0,"width":1920,"height":1080},"workArea":{"x":0,"y":0,"width":1920,"height":1080},"accelerometerSupport":"unknown","monochrome":false,"colorDepth":24,"colorSpace":"{primaries:BT709, transfer:SRGB, matrix:RGB, range:FULL}","depthPerComponent":8,"size":{"width":1920,"height":1080},"displayFrequency":60,"workAreaSize":{"width":1920,"height":1080},"scaleFactor":1,"rotation":0,"internal":false,"touchSupport":"unknown"}]},"ferdium":{"version":"6.0.0-nightly.101","electron":"19.0.8","installedRecipes":[{"id":"discord","version":"1.7.8"},{"id":"feedbin","version":"1.2.0"},{"id":"google-voice","version":"0.5.0"},{"id":"hangoutschat","version":"1.6.0"},{"id":"messenger","version":"1.7.3"},{"id":"msteams","version":"3.2.1"},{"id":"skype","version":"3.5.3"},{"id":"slack","version":"1.4.2"},{"id":"telegram","version":"3.3.4"},{"id":"whatsapp","version":"3.4.4"},{}],"devRecipes":[],"services":[{"id":"e3ecd545-6ed0-45ec-a73e-584d43eaf17d","recipe":"whatsapp","isAttached":true,"isActive":true,"isEnabled":true,"isHibernating":false,"hasCrashed":false,"isDarkModeEnabled":false,"isProgressbarEnabled":false},{"id":"0bd927e9-3d7b-4f73-97af-79bf29acd40d","recipe":"telegram","isAttached":true,"isActive":false,"isEnabled":true,"isHibernating":false,"hasCrashed":false,"isDarkModeEnabled":false,"isProgressbarEnabled":false},{"id":"7139e4aa-fb55-4798-b8a3-9152a261e73d","recipe":"hangoutschat","isAttached":true,"isActive":false,"isEnabled":true,"isHibernating":false,"hasCrashed":false,"isDarkModeEnabled":false,"isProgressbarEnabled":false},{"id":"4757334f-df87-4c03-89dc-704d36139be0","recipe":"slack","isAttached":true,"isActive":false,"isEnabled":true,"isHibernating":false,"hasCrashed":false,"isDarkModeEnabled":false,"isProgressbarEnabled":false},{"id":"1912bb5f-d253-4821-bd4d-d699913bbcf4","recipe":"discord","isAttached":true,"isActive":false,"isEnabled":true,"isHibernating":false,"hasCrashed":false,"isDarkModeEnabled":false,"isProgressbarEnabled":false},{"id":"47aaac1b-ad8f-4099-b52f-e15a3d5dc56d","recipe":"google-voice","isAttached":true,"isActive":false,"isEnabled":true,"isHibernating":false,"hasCrashed":false,"isDarkModeEnabled":false,"isProgressbarEnabled":false},{"id":"e792b72c-efc8-4ccc-be11-f07db830f4ee","recipe":"google-voice","isAttached":true,"isActive":false,"isEnabled":true,"isHibernating":false,"hasCrashed":false,"isDarkModeEnabled":false,"isProgressbarEnabled":false},{"id":"3bbffab9-da91-4da9-8c31-15c17bd2b236","recipe":"skype","isAttached":true,"isActive":false,"isEnabled":true,"isHibernating":false,"hasCrashed":false,"isDarkModeEnabled":false,"isProgressbarEnabled":false},{"id":"4f4afbd5-4684-4dc5-8a26-a217e5ce6995","recipe":"messenger","isAttached":true,"isActive":false,"isEnabled":true,"isHibernating":false,"hasCrashed":false,"isDarkModeEnabled":false,"isProgressbarEnabled":false},{"id":"8d3ae71a-d229-4e9b-92d5-ed5a5c811e16","recipe":"msteams","isAttached":false,"isActive":false,"isEnabled":false,"isHibernating":false,"hasCrashed":false,"isDarkModeEnabled":false,"isProgressbarEnabled":false}],"messages":[{"type":"error","info":["error","Warning: Failed %s type: %s%s","prop","Invalid prop `actions.settings` of type `Object` supplied to `Sidebar`, expected instance of `SettingsStore`.","\n    at Sidebar (/opt/Ferdium/resources/app.asar/components/layout/Sidebar.js:1:2062)\n    at /opt/Ferdium/resources/app.asar/node_modules/mobx-react/dist/mobxreact.cjs.development.js:524:34\n    at injectIntl(inject-with-stores-actions(Sidebar))\n    at div\n    at div\n    at ErrorBoundary (/opt/Ferdium/resources/app.asar/components/util/ErrorBoundary/index.js:1:822)\n    at injectIntl(ErrorBoundary)\n    at /opt/Ferdium/resources/app.asar/node_modules/react-jss/dist/react-jss.cjs.js:407:25\n    at AppLayout (/opt/Ferdium/resources/app.asar/components/layout/AppLayout.js:1:2699)\n    at /opt/Ferdium/resources/app.asar/node_modules/react-jss/dist/react-jss.cjs.js:407:25\n    at injectIntl(WithStyles(AppLayout))\n    at ThemeProvider (/opt/Ferdium/resources/app.asar/node_modules/theming/dist/theming.cjs.js:78:37)\n    at AppLayoutContainer (/opt/Ferdium/resources/app.asar/containers/layout/AppLayoutContainer.js:1:923)\n    at /opt/Ferdium/resources/app.asar/node_modules/mobx-react/dist/mobxreact.cjs.development.js:524:34\n    at Routes (/opt/Ferdium/resources/app.asar/node_modules/react-router/umd/react-router.development.js:887:7)\n    at Router (/opt/Ferdium/resources/app.asar/node_modules/react-router/umd/react-router.development.js:821:17)\n    at HistoryRouter (/opt/Ferdium/resources/app.asar/node_modules/react-router-dom/umd/react-router-dom.development.js:144:7)\n    at FerdiumRoutes (/opt/Ferdium/resources/app.asar/routes.js:1:2491)\n    at /opt/Ferdium/resources/app.asar/node_modules/mobx-react/dist/mobxreact.cjs.development.js:524:34\n    at IntlProvider (/opt/Ferdium/resources/app.asar/node_modules/react-intl/src/components/provider.js:84:47)\n    at I18N (/opt/Ferdium/resources/app.asar/I18n.js:1:506)\n    at /opt/Ferdium/resources/app.asar/node_modules/mobx-react/dist/mobxreact.cjs.development.js:524:34\n    at Provider (/opt/Ferdium/resources/app.asar/node_modules/mobx-react/dist/mobxreact.cjs.development.js:493:24)"]},{"type":"error","info":["error","Warning: Failed %s type: %s%s","prop","Invalid prop `actions.service` of type `Object` supplied to `QuickSwitchModal`, expected instance of `ServicesStore`.","\n    at QuickSwitchModal (/opt/Ferdium/resources/app.asar/features/quickSwitch/Component.js:1:2000)\n    at /opt/Ferdium/resources/app.asar/node_modules/mobx-react/dist/mobxreact.cjs.development.js:524:34\n    at /opt/Ferdium/resources/app.asar/node_modules/react-jss/dist/react-jss.cjs.js:407:25\n    at injectIntl(WithStyles(inject-with-stores-actions(QuickSwitchModal)))\n    at div\n    at div\n    at div\n    at ErrorBoundary (/opt/Ferdium/resources/app.asar/components/util/ErrorBoundary/index.js:1:822)\n    at injectIntl(ErrorBoundary)\n    at /opt/Ferdium/resources/app.asar/node_modules/react-jss/dist/react-jss.cjs.js:407:25\n    at AppLayout (/opt/Ferdium/resources/app.asar/components/layout/AppLayout.js:1:2699)\n    at /opt/Ferdium/resources/app.asar/node_modules/react-jss/dist/react-jss.cjs.js:407:25\n    at injectIntl(WithStyles(AppLayout))\n    at ThemeProvider (/opt/Ferdium/resources/app.asar/node_modules/theming/dist/theming.cjs.js:78:37)\n    at AppLayoutContainer (/opt/Ferdium/resources/app.asar/containers/layout/AppLayoutContainer.js:1:923)\n    at /opt/Ferdium/resources/app.asar/node_modules/mobx-react/dist/mobxreact.cjs.development.js:524:34\n    at Routes (/opt/Ferdium/resources/app.asar/node_modules/react-router/umd/react-router.development.js:887:7)\n    at Router (/opt/Ferdium/resources/app.asar/node_modules/react-router/umd/react-router.development.js:821:17)\n    at HistoryRouter (/opt/Ferdium/resources/app.asar/node_modules/react-router-dom/umd/react-router-dom.development.js:144:7)\n    at FerdiumRoutes (/opt/Ferdium/resources/app.asar/routes.js:1:2491)\n    at /opt/Ferdium/resources/app.asar/node_modules/mobx-react/dist/mobxreact.cjs.development.js:524:34\n    at IntlProvider (/opt/Ferdium/resources/app.asar/node_modules/react-intl/src/components/provider.js:84:47)\n    at I18N (/opt/Ferdium/resources/app.asar/I18n.js:1:506)\n    at /opt/Ferdium/resources/app.asar/node_modules/mobx-react/dist/mobxreact.cjs.development.js:524:34\n    at Provider (/opt/Ferdium/resources/app.asar/node_modules/mobx-react/dist/mobxreact.cjs.development.js:493:24)"]},{"type":"error","info":["error","Warning: Failed %s type: %s%s","prop","Invalid prop `actions.service` of type `Object` supplied to `PublishDebugLogModal`, expected instance of `ServicesStore`.","\n    at PublishDebugLogModal (/opt/Ferdium/resources/app.asar/features/publishDebugInfo/Component.js:1:2582)\n    at /opt/Ferdium/resources/app.asar/node_modules/mobx-react/dist/mobxreact.cjs.development.js:524:34\n    at /opt/Ferdium/resources/app.asar/node_modules/react-jss/dist/react-jss.cjs.js:407:25\n    at injectIntl(WithStyles(inject-with-stores-actions(PublishDebugLogModal)))\n    at div\n    at div\n    at div\n    at ErrorBoundary (/opt/Ferdium/resources/app.asar/components/util/ErrorBoundary/index.js:1:822)\n    at injectIntl(ErrorBoundary)\n    at /opt/Ferdium/resources/app.asar/node_modules/react-jss/dist/react-jss.cjs.js:407:25\n    at AppLayout (/opt/Ferdium/resources/app.asar/components/layout/AppLayout.js:1:2699)\n    at /opt/Ferdium/resources/app.asar/node_modules/react-jss/dist/react-jss.cjs.js:407:25\n    at injectIntl(WithStyles(AppLayout))\n    at ThemeProvider (/opt/Ferdium/resources/app.asar/node_modules/theming/dist/theming.cjs.js:78:37)\n    at AppLayoutContainer (/opt/Ferdium/resources/app.asar/containers/layout/AppLayoutContainer.js:1:923)\n    at /opt/Ferdium/resources/app.asar/node_modules/mobx-react/dist/mobxreact.cjs.development.js:524:34\n    at Routes (/opt/Ferdium/resources/app.asar/node_modules/react-router/umd/react-router.development.js:887:7)\n    at Router (/opt/Ferdium/resources/app.asar/node_modules/react-router/umd/react-router.development.js:821:17)\n    at HistoryRouter (/opt/Ferdium/resources/app.asar/node_modules/react-router-dom/umd/react-router-dom.development.js:144:7)\n    at FerdiumRoutes (/opt/Ferdium/resources/app.asar/routes.js:1:2491)\n    at /opt/Ferdium/resources/app.asar/node_modules/mobx-react/dist/mobxreact.cjs.development.js:524:34\n    at IntlProvider (/opt/Ferdium/resources/app.asar/node_modules/react-intl/src/components/provider.js:84:47)\n    at I18N (/opt/Ferdium/resources/app.asar/I18n.js:1:506)\n    at /opt/Ferdium/resources/app.asar/node_modules/mobx-react/dist/mobxreact.cjs.development.js:524:34\n    at Provider (/opt/Ferdium/resources/app.asar/node_modules/mobx-react/dist/mobxreact.cjs.development.js:493:24)"]},{"type":"error","request":{"result":null,"wasExecuted":true,"method":"getInfo"},"error":{},"response":{},"server":"https://api.ferdium.org"},{"type":"error","request":{"result":null,"wasExecuted":true,"method":"features"},"error":{},"response":{},"server":"https://api.ferdium.org"},{"type":"error","request":{"result":null,"wasExecuted":true,"method":"all"},"error":{},"response":{},"server":"https://api.ferdium.org"}],"workspaces":[],"windowSettings":{"width":1143,"height":860,"x":404,"y":88,"displayBounds":{"x":0,"y":0,"width":1920,"height":1080},"isMaximized":false,"isFullScreen":false},"settings":{"autoLaunchOnStart":false,"autoLaunchInBackground":false,"runInBackground":true,"reloadAfterResume":true,"reloadAfterResumeTime":10,"enableSystemTray":true,"startMinimized":false,"confirmOnQuit":false,"minimizeToSystemTray":false,"closeToSystemTray":false,"privateNotifications":false,"clipboardNotifications":true,"notifyTaskBarOnMessage":false,"showDisabledServices":true,"showServiceName":false,"showMessageBadgeWhenMuted":true,"showDragArea":false,"enableSpellchecking":true,"spellcheckerLanguage":"en-us","darkMode":false,"navigationBarManualActive":false,"splitMode":false,"splitColumns":3,"fallbackLocale":"en-US","beta":false,"isAppMuted":false,"enableGPUAcceleration":true,"enableGlobalHideShortcut":false,"server":"https://api.ferdium.org","predefinedTodoServer":"https://todoist.com/app","autohideMenuBar":false,"lockingFeatureEnabled":false,"locked":false,"lockedPassword":"******","useTouchIdToUnlock":true,"scheduledDNDEnabled":false,"scheduledDNDStart":"17:00","scheduledDNDEnd":"09:00","hibernateOnStartup":true,"hibernationStrategy":300,"wakeUpStrategy":300,"wakeUpHibernationStrategy":0,"wakeUpHibernationSplay":true,"inactivityLock":0,"automaticUpdates":true,"universalDarkMode":true,"userAgentPref":"","adaptableDarkMode":true,"accentColor":"#7367F0","progressbarAccentColor":"#7367F0","serviceRibbonWidth":55,"sidebarServicesLocation":0,"iconSize":20,"navigationBarBehaviour":"custom","searchEngine":"startPage","useHorizontalStyle":true,"hideCollapseButton":false,"isMenuCollapsed":false,"hideRecipesButton":false,"hideSplitModeButton":true,"useGrayscaleServices":false,"grayscaleServicesDim":50,"hideWorkspacesButton":false,"hideNotificationsButton":false,"hideSettingsButton":false,"alwaysShowWorkspaces":false,"liftSingleInstanceLock":false,"enableLongPressServiceHint":false,"proxyFeatureEnabled":false,"onlyShowFavoritesInUnreadCount":false,"customTodoServer":"","locale":"en-US","useVerticalStyle":true},"features":{},"user":"82c1cf9d-ab58-4da2-b55e-aaa41d2142d8"}}

### Additional information

sorry I should have logged this when it first occurred. Not sure what version or build this was working on as I frequently move between PCs and OSs throughout the week. Devs - thanks for your work on this project. This program has become as invaluable as my browser.

**Comment by SpecialAro** [comment]
Hello @fgurdradee , thank you for opening this issue.

As discussed [here](https://www.reddit.com/r/ferdium/comments/w09jgh/comment/igir3sm/?utm_source=share&utm_medium=web2x&context=3), this happens because we hardcoded the Dark Mode for the Whatsapp recipe.

If you can, right click on the recipe and click enable dark mode. This way, Ferdium will tell WhatsApp to use darkmode.

You might need to reload the recipe after changing this setting with `Ctrl/CMD + R`.

Please let me know if I can close this issue 😄

**Comment by fgurdradee** [comment]
Thank you @SpecialAro for your help and the reddit link. 

Longer term, I'm sure this will be logged as an issue or in forums from other users when we get to final and more people start using Ferdium. 

Is hardcoding dark mode the best way to do this? 

What was the recipe using before? Light/Dark/System theme? 

Thanks again. Feel free to close but please consider the user experience consequences of this.

**Comment by krisrok** [comment]
Stumbled upon this too. I guess the best would be to always synchronize the hardcoded setting and the whatsapp setting, if at all possible: If hardcoded setting changes -> write back setting to whatsapp settings. If whatsapp settings change -> write change back to hardcoded setting.

**Comment by OpenSauce04** [comment]
I just had to look this up because I was experiencing the same thing. I never saw the warning which was introduced in ferdium/ferdium-recipes#138, instead just being left to experience what seemed to be a very annoying bug. (edit, seems like this warning was reverted due to it causing issues 🙃)

Why is the dark mode for WhatsApp hardcoded? Is there some technical reason why it is that way? If not, this just seems like a horrible UX decision.
